include/seqan3/core/simd/view_iota_simd.hpp

```diff
@@ -0,0 +1,248 @@
+ * | std::ranges::contiguous_range    | *lost*                         |
+ * |                                  |                                |
+ * | std::ranges::viewable_range      | *guaranteed*                   |
+ * | std::ranges::view                | *guaranteed*                   |
+ * | std::ranges::sized_range         | *guaranteed*                   |
+ * | std::ranges::common_range        | *guaranteed*                   |
+ * | std::ranges::output_range        | *lost*                         |
+ * | std::ranges::borrowed_range      | *guaranteed*                   |
+ * | seqan3::const_iterable_range     | *guaranteed*                   |
+ * |                                  |                                |
+ * | std::ranges::range_reference_t   | simd_t                         |
+ *
+ * For more details, see \ref views.
+ *
+ * ### Example
+ *
+ * \include test/snippet/core/simd/view_iota_simd.cpp
+ */
+template <simd_concept index_t>
+class iota_simd : public std::ranges::view_interface<iota_simd<index_t>>
+{
+private:
+    //!\brief The underlying scalar type.
+    using scalar_type = typename simd_traits<index_t>::scalar_type;
+    //!\brief The counted simd iterator type.
+    using iterator_type = seqan3::detail::counted_simd_iterator<index_t>;
+
+    //!\brief The begin index.
+    scalar_type begin_index{};
+    //!\brief The end index.
+    scalar_type end_index{};
+
+public:
+    /*!\name Constructor, assignment and destructor
+     * \{
+     */
+    iota_simd() noexcept = default; //!< Defaulted.
+    iota_simd(iota_simd const &) noexcept = default; //!< Defaulted.
+    iota_simd(iota_simd &&) noexcept = default; //!< Defaulted.
+    iota_simd & operator=(iota_simd const &) noexcept = default; //!< Defaulted.
+    iota_simd & operator=(iota_simd &&) noexcept = default; //!< Defaulted.
+    ~iota_simd() = default; //!< Defaulted.
+
+    /*!\brief Constructs the iota view from the given index pair.
+     *
+     * param[in] begin_index The first index of the iota view.
+     * param[in] end_index The index behind the last one of the iota view.
+     */
+    iota_simd(scalar_type const begin_index, scalar_type const end_index) :
+        begin_index{begin_index},
+        end_index{end_index}
+    {}
+    //!\}
+
+    /*!\name Iterators
+     * \{
+     */
+    //!\brief Returns seqan3::detail::counted_simd_iterator pointing to the begin of the range.
+    iterator_type begin() const noexcept
+    {
+        return iterator_type{begin_index};
+    }
+
+    //!\brief Returns seqan3::detail::counted_simd_iterator pointing to the end of the range.
+    iterator_type end() const noexcept
+    {
+        return iterator_type{end_index};
+    }
+    //!\}
+};
+
+} // namespace seqan3::views
+
+//!\cond
+// Enable borrowed range.
+// namespace std::ranges
+// {
+// template <typename index_t>
+// inline constexpr bool enable_borrowed_range<seqan3::views::iota_simd<index_t>> = true;
+// }
+//!\endcond
```

Comment:
This should have an issue; Instead of this TODO in the code

Reply:
that is quite right.